src/main/java/CommunicationChannel.java

```diff
@@ -0,0 +1,21 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class CommunicationChannel {
+
+    private final OutputStream output;
+    private final InputStream input;
+
+    public CommunicationChannel(InputStream input, OutputStream output) {
+        this.input = input;
+        this.output = output;
+    }
+
+    public OutputStream getOutput() {
```

Comment:
I would expect this method to be named `getOutputStream` and similarly for the input.